pwa/src/components/welcomeModal/welcomeModal.tsx

```diff
@@ -0,0 +1,57 @@
+
+import * as React from "react";
+import { navigate } from "gatsby";
+import APIService from "../../apiService/apiService";
+import APIContext from "../../apiService/apiContext";
+import { Modal } from "@conductionnl/nl-design-system";
+
+const WelcomeModal: React.FC = () => {
+  const API: APIService = React.useContext(APIContext);
+  const buttonRef = React.useRef(null);
+
+  React.useEffect(() => {
+    // Check route
+    if (window.location.href.split("/").at(-1) !== 'applications' || (window.location.href.split("/").at(-1) !== 'new' && window.location.href.split("/").at(-2) !== 'applications')) {
```

Comment:
I think this if statement got more complex than necessary, the following does the same:

`window.location.pathname.includes('applications') // results in a true or false depending if 'applications' is in the path`

Above in mind, I think the following set-up might be a bit better for readability and in a more consistent React way of working:

```
...
  const [applicationsCount, setApplicationsCount] = React.useState<number>(null)
  const API: APIService = React.useContext(APIContext);
  const buttonRef = React.useRef(null);

  React.useEffect(() => { handleSetApplicationsCount() }, [API])

  React.useEffect(() => {
    if (applicationsCount === 0 || !buttonRef.current || window.location.pathname.includes('applications')) return

    buttonRef.current.click()
  }, [buttonRef])

  const handleSetApplicationsCount = (): void => {
    API.Application.getAll()
      .then((res) => { setApplicationsCount(res.data.length) })
      .catch((err) => { throw new Error('GET applications error: ' + err) })
  }
...
```